### PR TITLE
Added DisplayAlert Support

### PR DESCRIPTION
**Pull request by ahoefling**
Added basic DisplayAlert Support.

The DisplayAlert API in Xamarin.Forms uses the `MessagingCenter` to communicate with platform code. This is a simple `Subscribe` and `Unsubscribe` including adding and removing the element from the Renderer. 

image

**Usage**
```c#
public async void OnButtonPressed(object sender, EventArgs e)
{
    await DisplayAlert("Title", "Some message", "OK");
}
```

**Styles**
Following inline with our current UI styles I used Bootstrap styles to build the modal

**Current Features**
* Displays modal using boostrap styles
* Title
* Message
* Cancel Button (typically your OK button)
    * This means it currently does not support 2 buttons, right now it is just 1 button
* X button at the top right corner of the modal
* Closing the modal by clicking the cancel button or the x button

**Still needs implementing**
A list of items that need to be implemented still

* When the API is called with both `accept` and `cancel` it needs to return true/false depending on what the user clicks

--
This is where I finished up after playing around with it tonight. I wanted to submit the PR so we can have the code and provide any feedback. I plan to finish up this PR with implementing the return value logic.

### CODE REVIEW

**Comment by praeclarum** [comment]
This is great! I was debating whether to use `alert()` itself, but this looks to be a better approach.

My biggest comment so far has only to do with code formatting. Please make sure your editor obeys the .editorconfig so that all the reformats in Platform.cs go away.

The code looks good now. If you want to keep this PR open to finish that last feature you mentioned, that's fine. Otherwise I can merge and another PR can finish it. Let me know what you prefer.

Try to get a commit in that fixes the formatting if you can. Thanks!

**Comment by ahoefling** [comment]
I was thinking the exact same thing of using alert vs bootstrap but since we are using bootstrap all over the place I thought it would be a good place to start. Granted if the bootstrap stylesheet is not referenced this will **not** work. Perhaps we should add some custom styles to make sure this always works. Unless we are saying Ooui.Forms is going to be dependent on bootstrap.

**Code Formatting** 
* I have no idea how the code formatting code like this, I'll take care of it and update the PR

**Additional Changes**
* Add support for accept/cancel bool return values
* Clean up modal code and move it into it's own class so it is easier to manage

** Build failing**
* Is my current PR okay as far as a build goes? 
* I notice the bitrise build is failing but can't see the output

**Comment by praeclarum** [comment]
For now I'm OK to take a dependency on bootstrap. I kept wavering, but I think it's the smart way to go.

In the future, it would be nice to integrate other UI libraries (semantic and material come to mind) - but we'll tackle that when we need it.

**Comment by ahoefling** [comment]
@praeclarum regarding the formatting issue: there is an issue with the Platform.cs and it's line endings there is a mismatch of `LF` and `CRLF`. It would be best to normalize the line endings, how would you like this done? 

image

**Comment by ahoefling** [comment]
That was a mild nightmare to fix but I got the line endings all cleaned up in the `Platform.cs` class. Now I'll work on the rest of my changes

File: Ooui.Forms/Platform.cs [hunk 1]
```diff
@@ -31,6 +32,19 @@ public class Platform : BindableObject, IPlatform, INavigation, IDisposable
         public Platform ()
         {
             _renderer = new PlatformRenderer (this);
+
+            MessagingCenter.Subscribe(this, Page.AlertSignalName, (Page sender, AlertArguments arguments) =>
+            {
```
Comment (ahoefling) [inline Ooui.Forms/Platform.cs]: We really should put a check into here to return if there is already a DisplayAlert on the screen. I tried adding a simple bool into the platform to handle the state but this ran into a race condition. If you have a second DisplayAlert that displays based on the result of the first the 2nd display alert will not display due to this race condition

**Comment by ahoefling** [comment]
@praeclarum I would say the PR is ready to merge into master. I added a code file comment regarding an issue with multiple DisplayAlert's rendering at the same time and a solution I tried that didn't work.

Let me know if you need me to do more work on this

**Comment by praeclarum** [comment]
Looks great, thanks again for your hard work!

**Pull request merged**